components/director/internal/domain/application/repository.go

```diff
@@ -57,12 +67,71 @@ func (r *inMemoryRepository) List(ctx context.Context, tenant string, filter []*
 	}, nil
 }
 
-// TODO: Make filtering and paging
-func (r *inMemoryRepository) ListByRuntimeID(ctx context.Context, tenant, runtimeID string, pageSize *int, cursor *string) (*model.ApplicationPage, error) {
+// TODO: @dbadura add pagination when PR-181 is merged
+func (r *inMemoryRepository) ListByScenariosForRuntime(ctx context.Context, tenantID string, runtimeID string, pageSize *int, cursor *string) (*model.ApplicationPage, error) {
```

Comment:
I would go a different way. This method should return applications by filters. The filters should be built on the service or even upper level. IMO This repository should not have the notion of runtimes. @aszecowka what do you think?

Reply:
Introduced new method name `ListByScenarios`.